src/librustc_trans/back/link.rs

```diff
@@ -1077,6 +1117,11 @@ fn link_args(cmd: &mut Linker,
     cmd.args(&used_link_args);
 }
 
+// Checks if sanitizer supports position independent executables.
+fn sanitizer_support_pie(sess: &Session) -> bool {
+    sess.opts.cg.sanitize != Some(Sanitize::Thread)
```

Comment:
Could you add a comment why the thread sanitizer *doesn't* support position independent executables?